src/prototype.rs

```diff
@@ -210,6 +241,86 @@ impl<'a> FirstPass<'a> {
         ix
     }
 
+    /// When start_ix is at the beginning of an HTML block of type 1 to 5,
+    /// this will find the end of the block, adding the block itself to the
+    /// tree and also keeping track of the lines of HTML within the block.
+    ///
+    /// The html_end_tag is the tag that must be found on a line to end the block.
+    fn parse_html_block_type_1_to_5(&mut self, start_ix: usize, html_end_tag: &'static str) -> usize {
+        self.tree.append(Item {
+            start: start_ix,
+            end: 0, // set later
+            body: ItemBody::HtmlBlock(Some(html_end_tag)),
+        });
+        self.tree.push();
+
+        let mut ix = start_ix;
+        let end_ix;
+        loop {
+            let line_start_ix = ix;
+            ix += scan_nextline(&self.text[ix..]);
+            self.append_html_line(line_start_ix, ix);
+
+            let mut line_start = LineStart::new(&self.text[ix..]);
+            let n_containers = self.scan_containers(&mut line_start);
+            if n_containers < self.tree.spine.len() {
+                end_ix = ix;
+                break;
+            }
+
+            if (&self.text[line_start_ix..ix]).contains(html_end_tag) {
+                end_ix = ix;
+                break;
+            }
+
+            let next_line_ix = ix + line_start.bytes_scanned();
```

Comment:
I think you want to advance `ix` here. If last line doesn't end in a newline, then I think `ix` will be left at the _beginning_ of that line on return.

Reply:
I had copied this part from the `parse_indented_code_blocks()`. It does seem correct to advance it here. 

Do you recall why you didn't want to advance `ix` in the similar case there?

Reply:
I misunderstood the first time I read this, thought there was a line that was to be included. I think the code is ok here, but probably want to set `end_ix` here for sourcemap. Actually I'm not sure you need a separate `end_ix`, it's good enough to `self.pop(ix)` at the end. The reason the indented code blocks need that is that blank line is not considered part of the block (so `end_ix` points before the blank line and `ix` after).

Reply:
Let's just make sure I haven't blundered here as well:

```
<pre></pre>

Text
```

Gives an `HtmlBlock(0, 12)` and I believe that means bytes 0-11 are in the block and byte 12 is not in the block? So the HtmlBlock contains from the opening `<` until the newline at the end of that line, but not the newline for the blank line.

Does that seem right?